src/PackageDescription.ts

```diff
@@ -9,36 +9,96 @@ export type PackageDescription = {
 
   /** Filter of package versions to collect and show */
   versionFilter: (version: string) => boolean;
+
+  /** Allows relabeling specific versions to a user-friendly value */
+  versionLabeler?: (version: string) => string;
 };
 
+function canaryVersionLabeler(version: string): string {
+  if (version === "0.0") {
+    return "canary";
+  }
+
+  if (version.startsWith("0.0.0-canary")) {
+    return version.slice(6, version.length);
+  }
+
+  return version;
+}
+
+function nightlyHashVersionLabeler(version: string): string {
+  if (version === "0.0") {
+    return "nightly";
+  }
+
+  if (version.match(/^0\.0\.0-[0-9a-f]+$/)) {
+    const hash = version.slice(6);
+    const shortenedHash = hash.slice(0, Math.min(hash.length, 7));
+
+    return `nightly@${shortenedHash}`;
+  }
```

Comment:
If you're not using the result, you should use `test()` instead:
```ts
  if (/^0\.0\.0-[0-9a-f]+$/.test(version)) {
```
Alternatively, if we're already using regex, you can also capture the hash:
```suggestion
  const m = version.match(/^0\.0\.0-([0-9a-f]{0,7})[0-9a-f]*$/);
  if (m) {
    return `nightly@${m[1]}`;
  }
```